backend/src/schema/resolvers/emails.js

```diff
@@ -0,0 +1,92 @@
+import generateNonce from './helpers/generateNonce'
+import Resolver from './helpers/Resolver'
+import existingEmailAddress from './helpers/existingEmailAddress'
+import { UserInputError } from 'apollo-server'
+import Validator from 'neode/build/Services/Validator.js'
+
+export default {
+  Mutation: {
+    AddEmailAddress: async (_parent, args, context, _resolveInfo) => {
+      let response
+      try {
+        const { neode } = context
+        await new Validator(neode, neode.model('UnverifiedEmailAddress'), args)
+      } catch (e) {
+        throw new UserInputError('must be a valid email')
+      }
+
+      // check email does not belong to anybody
+      await existingEmailAddress(_parent, args, context)
+
+      const nonce = generateNonce()
+      const {
+        user: { id: userId },
+      } = context
+      const { email } = args
+      const session = context.driver.session()
+      const writeTxResultPromise = session.writeTransaction(async txc => {
+        const result = await txc.run(
+          `
+            MATCH (user:User {id: $userId})
+            MERGE (user)<-[:BELONGS_TO]-(email:UnverifiedEmailAddress {email: $email, nonce: $nonce})
+            SET email.createdAt = toString(datetime())
+            RETURN email, user
+          `,
+          { userId, email, nonce },
+        )
+        return result.records.map(record => ({
+          name: record.get('user').properties.name,
+          ...record.get('email').properties,
+        }))
+      })
+      try {
+        const txResult = await writeTxResultPromise
+        response = txResult[0]
+      } finally {
+        session.close()
+      }
+      return response
+    },
+    VerifyEmailAddress: async (_parent, args, context, _resolveInfo) => {
+      let response
+      const {
+        user: { id: userId },
+      } = context
+      const { nonce, email } = args
+      const session = context.driver.session()
+      const writeTxResultPromise = session.writeTransaction(async txc => {
+        const result = await txc.run(
```

Comment:
`const result` would be a result set. It's a local variable to store that stuff we need for the return value :laughing:

Reply:
I get that, just think we don't really gain anything by being vague, and it could make our code easier to understand if it's more expressive rather than less.